lib/data-table/table.rb

```diff
@@ -284,10 +284,19 @@ def render_subtotals(group_header, _group_data = nil, ancestors = nil)
       html = ''
       path = ancestors.nil? ? [] : ancestors.dup
       path << group_header
+
+      first_rendered_subtotal = nil
+
       @subtotal_calculations[path].each_with_index do |group, index|
+        if first_rendered_subtotal.nil? && !group.empty?
+          first_rendered_subtotal = 'first'
+        else
```

Comment:
This logic feels weird. What is the value of `first_rendered_subtotal` for the second subtotal row? The logic to reset to `nil` will only run if first_rendered_subtotal is still `nil`. Is there a test case for this?

Reply:
It's `nil` for any row that isn't the first rendered one.  I modified an existing test in this PR, see changes below this comment.

#### Example Data Table with 2 subtotal rows

```ruby
collection = [
  {:name => 'Luke Skywalker', :class => 'Jedi Knight', :world => 'Star Wars', :power_level => 50},
  {:name => 'Emporer Palpatine', :class => 'Sith Lord', :world => 'Star Wars', :power_level => 95},
  {:name => 'Mithrander', :class => 'Wizard', :world => 'Middle Earth', :power_level => 9001},
  {:name => 'Aragorn', :class => 'Ranger', :world => 'Middle Earth', :power_level => 80}
]

DataTable.render(collection) do |t|
  t.column :power_level, "Power Level"
  t.column :mana, "Mana"

  t.group_by :world

  # two subtotal rows
  t.subtotal :power_level, :avg, 10 do |sub|
    "subtotal avg: #{sub}"
  end
  t.subtotal :mana, :avg, 11 do |sub|
    "subtotal avg: #{sub}"
  end
end
```

#### Generated HTML

```html


<table id="" class="data_table " cellspacing="0" cellpadding="0">
   <caption></caption>
   <thead>
      <tr>
         <th class="power_level ">Power Level</th>
         <th class="mana ">Mana</th>
      </tr>
   </thead>
   <tbody class="star_wars">
      <tr class="group_header level_0">
         <th colspan="2">Star Wars</th>
      </tr>
      <tr class="row_0 ">
         <td class="power_level numeric">50</td>
         <td class="mana nilclass"></td>
      </tr>
      <tr class="row_1 alt ">
         <td class="power_level numeric">95</td>
         <td class="mana nilclass"></td>
      </tr>
      <!-- first rendered row, gets top border -->
      <tr class="subtotal index_10 first">
         <td class="power_level numeric">subtotal avg: 72.5</td>
         <td class="mana nilclass"></td>
      </tr>
      <!-- second rendered row, does not get top border -->
      <tr class="subtotal index_11 ">
         <td class="power_level numeric"></td>
         <td class="mana nilclass">subtotal avg: 0.0</td>
      </tr>
   </tbody>
   <tbody class="middle_earth">
      <tr class="group_header level_0">
         <th colspan="2">Middle Earth</th>
      </tr>
      <tr class="row_0 ">
         <td class="power_level numeric">9001</td>
         <td class="mana nilclass"></td>
      </tr>
      <tr class="row_1 alt ">
         <td class="power_level numeric">80</td>
         <td class="mana nilclass"></td>
      </tr>
      <tr class="subtotal index_10 first">
         <td class="power_level numeric">subtotal avg: 4540.5</td>
         <td class="mana nilclass"></td>
      </tr>
      <tr class="subtotal index_11 ">
         <td class="power_level numeric"></td>
         <td class="mana nilclass">subtotal avg: 0.0</td>
      </tr>
   </tbody>
</table>


```

Reply:
Interesting. Well, it it works, it works. It's hard to deduce from this change that it would work though, but that's probably just because I'm not very familiar with this code base.


Is it possible to move this if statement below the `next if group.empty?` line below? Seems odd to do 2 group.empty checks back to back.